server/src/main/java/org/cloudfoundry/identity/uaa/provider/oauth/OauthIDPWrapperFactoryBean.java

```diff
@@ -61,6 +61,16 @@ else if(OIDC10.equalsIgnoreCase(type)) {
                         oidcIdentityProviderDefinition.setPasswordGrantEnabled(idpDefinitionMap.get("passwordGrantEnabled") == null ? false : (boolean) idpDefinitionMap.get("passwordGrantEnabled"));
                         oidcIdentityProviderDefinition.setSetForwardHeader(idpDefinitionMap.get("setForwardHeader") == null ? false : (boolean) idpDefinitionMap.get("passwordGrantEnabled"));
                         oidcIdentityProviderDefinition.setPrompts((List<Prompt>) idpDefinitionMap.get("prompts"));
+                        if (idpDefinitionMap.get("jwtclientAuthentication") != null) {
```

Comment:
The logic in this method is already too deeply nested. Please extract some methods to simplify the code.

Reply:
done